guardian/itunes-rss
# Return 404s not 500s when podcasts or tags are not found

## What does this change?
We occasionally see alerts going off for high 5xx rates from the itunes-rss load balancer. On closer inspection it seems that any non-success code (usually 404s) were being returned as 500s.

This PR extends the `Good` `Or` `Bad` model a little by enclosing a `Failed` object in a `Bad` result that contains a message and status code that can be logged properly, and prevent 500s being returned when we're merely being asked for non-existent resources.

A new test is added using an example picked from live logs to check that the expected flavour of `Bad` is generated when the tag doesn't have a podcast field.

## How to test
We only have a PROD environment for this - so I'm hoping the unit test is sufficient 😬  Once deployed though we can  check manually but expect to see the culprit(s) asking for non-existent data and thus triggering the new code.

## How can we measure success?
We shouldn't return 5xx codes when we really mean 404

## Have we considered potential risks?
The podcast.xml endpoint may break. We can roll back if that happens.

## Images
N/A

## Accessibility
N/A
